packages/headless/src/controllers/core/generated-answer/headless-answer-stream-manager.ts

```diff
@@ -0,0 +1,53 @@
+import {GeneratedAnswerAPIClient} from '../../../api/generated-answer/generated-answer-client';
+import {CoreEngine} from '../../../app/engine';
+import {ClientThunkExtraArguments} from '../../../app/thunk-extra-arguments';
+import {
+  notifyStreamAborted,
+  registerAnswerStreamManager,
+  streamAnswer,
+} from '../../../features/generated-answer/generated-answer-actions';
+import {
+  GeneratedAnswerSection,
+  SearchSection,
+  DebugSection,
+} from '../../../state/state-sections';
+
+type AnswerStreamManagerEngine = CoreEngine<
+  GeneratedAnswerSection & SearchSection & DebugSection,
+  ClientThunkExtraArguments<GeneratedAnswerAPIClient>
+>;
+
+export function buildAnswerStreamManager(engine: AnswerStreamManagerEngine) {
+  if (engineHasAnswerStreamManager(engine)) {
+    return;
+  }
+
+  let abortController: AbortController | undefined;
+  const setAbortControllerRef = (ref: AbortController) => {
+    abortController = ref;
+  };
+
+  function abortStream() {
+    abortController?.abort();
+    abortController = undefined;
+  }
+
+  engine.dispatch(registerAnswerStreamManager);
+  engine.subscribe(() => {
+    if (engine.state.generatedAnswer.shouldStartStream) {
+      abortStream();
+      engine.dispatch(
+        streamAnswer({
+          setAbortControllerRef,
+        })
+      );
```

Comment:
I think `shouldStartStream`  should be reset to `false` in the state after dispatching the `streamAnswer` action.

Otherwise we would just keep streaming answers infinitely for every change that occurs on the state.